sqlalchemy_ibmi/base.py

```diff
@@ -728,11 +728,17 @@ def create_connect_args(self, url):
         if 'port' in opts and opts['port'] is None:
             opts.pop('port')
         allowed_opts = {'host', 'user', 'password',
-                        'autocommit', 'readonly', 'timeout','database'}
+                        'autocommit', 'readonly', 'timeout', 'database'}
+        if 'database' in opts and opts['database'] is None:
+            opts.pop('database')
+        if 'database' not in opts:
+            opts.update({'dsn': opts.pop('host')})
+            allowed_opts.add('dsn')
+            allowed_opts.difference({'host', 'database'})
```

Comment:
This is not really correct. We can set DSN _and_ database/host/etc as well. Let's stick to non-DSN support for now, though, since I envision DSN work differently.